File: services/engagement-svc/src/routes/health.ts
import { FastifyInstance } from "fastify";
import { healthSchema } from "./schemas.js";

export function registerHealthRoutes(app: FastifyInstance) {
  const handler = async () => ({
    status: "ok",
    service: "engagement-svc",
    timestamp: new Date().toISOString(),
  });

  app.get("/api/engagement/health", { schema: healthSchema }, handler);
  // Liveness/Readiness/Startup probes hit /health
  app.get("/health", { schema: healthSchema }, handler);
}

File: services/ai-svc/src/ai_svc/routes/health.py
from fastapi import APIRouter

router = APIRouter(prefix="/api/ai", tags=["health"])

# Root-level router exposed for k8s liveness/readiness/startup probes that
# hit /health (without the /api/ai prefix).
root_router = APIRouter(tags=["health"])


@router.get("/health")
async def health():
    return {"status": "ok", "service": "ai-svc"}


@root_router.get("/health")
async def health_root():
    return {"status": "ok", "service": "ai-svc"}
